### PR TITLE
fix: use mean() instead of sum() for contrastive loss in SimGCL and NCL

### LINKED ISSUE #91: [🐛BUG] SimGCL的对比损失计算错误。
**描述这个 bug**
SimGCL的对比损失计算错误。
按照我对论文的理解，应该是单个对象（user或item）的计算对比损失，但是代码中应该是将一个mini batch的所有对比损失进行了求和。

**如何复现**
在simgcl.py的函数calculate_cl_loss中，最后将所有的对比损失进行的是求和操作。代码如下：
return -torch.log(pos_score / ttl_score).sum()
但是这样做会将一个mini batch的所有损失进行求和，而它继承的LightGCN在计算损失时，使用的均值。
这样会导致两个损失的计算维度不一致，从而使模型精度下降。

**预期**
计算对比损失也是用均值。即：
return -torch.log(pos_score / ttl_score).mean()

我本机实验，采用均值的效果会远好于求和，可以复现论文中的效果，求和则差距很大。

### PR DESCRIPTION
**Pull request by WeishuZ**
Fixes #91

## Problem
The contrastive loss in SimGCL and NCL used `.sum()` to aggregate the InfoNCE loss, while the base BPR loss (from LightGCN/RecBole) uses `.mean()`. This scale mismatch causes the CL loss to dominate by a factor of batch_size, leading to failure to reproduce paper results.

## Fix
Changed `.sum()` → `.mean()` in:
- **SimGCL**: `calculate_cl_loss()` (1 occurrence)
- **NCL**: `proto_nce_loss` and `ssl_loss` for both users and items (4 occurrences)

## Consistency
This aligns with other CL-based models in the repo that already use `.mean()`:
- LightGCL ✅
- SSL4Rec ✅
-  XSimGCL ✅

As reported in #91, using `.mean()` reproduces the original paper results, while `.sum()` gives significantly worse performance.

### CODE REVIEW

**Comment by dst111dst** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我最近正在休假中，无法亲自回复你的邮件。我将在假期结束后，尽快给你回复。图片链接二维码图片
